backend/handlers.py

```diff
@@ -722,6 +724,23 @@ def post(self):
 
   options = util.EnableCors
 
+# A handler for monitoring progress towards matching from the Presidential
+# Election Campaign Fund: http://www.fec.gov/press/bkgnd/fund.shtml
+# "... raising at least $5,000 in each of 20 states. Only contributions from
+# individuals, and only contributions up to $250, are matchable"
+class StatesHandler(webapp2.RequestHandler):
+  def get(self):
+    util.EnableCors(self)
+    totals = {}
+    for pledge in model.Pledge.all():
+      user = model.User.all().filter('email =', pledge.email).get()
+      totals[user.state] += min(pledge.amountCents, 25000) # max $250
+
+    for state, total in totals.iteritems():
+      totals[state] = min(500000, total) # no point counting past $5k
```

Comment:
Showing the total is fine, as long as we are limiting each donation to 250
(qualified amount for federal matching funds)

A


On Tue, Sep 15, 2015 at 11:41 AM, brafkind <notifications@github.com> wrote:

> In backend/handlers.py
> <https://github.com/Lessig2016/pledgeservice/pull/4#discussion_r39526317>:
>
> > @@ -722,6 +724,23 @@ def post(self):
> >
> >    options = util.EnableCors
> >
> > +# A handler for monitoring progress towards matching from the Presidential
> > +# Election Campaign Fund: http://www.fec.gov/press/bkgnd/fund.shtml
> > +# "... raising at least $5,000 in each of 20 states. Only contributions from
> > +# individuals, and only contributions up to $250, are matchable"
> > +class StatesHandler(webapp2.RequestHandler):
> > +  def get(self):
> > +    util.EnableCors(self)
> > +    totals = {}
> > +    for pledge in model.Pledge.all():
> > +      user = model.User.all().filter('email =', pledge.email).get()
> > +      totals[user.state] += min(pledge.amountCents, 25000) # max $250
> > +
> > +    for state, total in totals.iteritems():
> > +      totals[state] = min(500000, total) # no point counting past $5k
>
> Is it actually strategically advantageous to limit the funds total to $5k
> per state? This campaign also espouses transparency. What would be the harm
> of showing the actual total?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Lessig2016/pledgeservice/pull/4/files#r39526317>.
>